apps/language_server/lib/language_server/build.ex

```diff
@@ -20,15 +20,14 @@ defmodule ElixirLS.LanguageServer.Build do
                 IO.puts("MIX_TARGET: #{apply(Mix, :target, [])}")
               end
 
-              prev_deps = cached_deps()
-              :ok = Mix.Project.clear_deps_cache()
-
               case reload_project() do
                 {:ok, mixfile_diagnostics} ->
                   # FIXME: Private API
                   if Keyword.get(opts, :fetch_deps?) and
-                       Mix.Dep.load_on_environment([]) != prev_deps,
-                     do: fetch_deps()
+                       Mix.Dep.load_on_environment([]) != cached_deps() do
+                    :ok = Mix.Project.clear_deps_cache()
```

Comment:
Can we add an explanatory comment about why this needs to be within the if statement?

Reply:
@axelson NOTE comment added.